src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,248 @@
+            2. Generate SAS for the storage account selected\provided or default supplied.
+            3. Run the clean-up and then aks-persicope on the cluster with current cluster kubeconfig.
+            4. Generate Dowloadable link once action is performed at the webView.
+        **/
+        await longRunning(`Checking storage associated cluster.`,
+          async () => {
+            // Check if Diagnostic setting exist.
+            const diagnosticSettings = await getClusterDiagnosticSettings(cloudResource);
+
+            if (diagnosticSettings?.value?.length) {
+              // If cluster diagnosticSetting exist pass it on for UI workflow and aks-preiscope deployment.
+              await selectStorageAccountAndInstallAKSPeriscope(diagnosticSettings, clusterKubeConfig, cloudResource);
+            } else if (!diagnosticSettings?.value?.length) {
+              // If there is no Diagnostic setting - Present UI for user input.
+              await getStorageAccountNameAndKeyUserInputAndInstallAKSPeriscope(clusterKubeConfig, cloudResource.name);
+            }
+          });
+      }
+    } else {
+      vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+    }
+  }
+}
+
+async function selectStorageAccountAndInstallAKSPeriscope(
+  diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection | undefined,
+  clusterKubeConfig: string,
+  cloudResource: any
+) {
+
+  let periscopeStorageInfo: PeriscopeStorage | undefined;
+  const clusterName = cloudResource.name;
+
+  /*
+      Check the diagnostic setting is 1 or more than 1:
+        1. For the scenario of 1 - Pick the storageId resource and get SAS.
+        2. For the scenario for more than 1 then show VsCode quickPick to select and get SAS of selected.
+  */
+
+  if (diagnosticSettings && diagnosticSettings?.value!.length > 1) {
+    const storageAccountNameToStorageIdMap: Map<string, string> = new Map();
+
+    diagnosticSettings.value?.forEach((item): any => {
+      if (item.storageAccountId) {
+        const { name } = parseResource(item.storageAccountId!);
+        if (!name) {
+          vscode.window.showInformationMessage(`Storage Id is malformed: ${item.storageAccountId}`);
+          return undefined;
+        }
+        storageAccountNameToStorageIdMap.set(name!, item.storageAccountId!);
+      }
+    });
+
+    const quickPick = vscode.window.createQuickPick();
+    quickPick.placeholder = "Select storage account for periscope deployment:";
+    quickPick.title = `Found more than 1 storage account associated with cluster ${clusterName}. Please select from below.`;
+    quickPick.ignoreFocusOut = true;
+    quickPick.items = Array.from(storageAccountNameToStorageIdMap.keys()).map((label) => ({ label }));
+    quickPick.onDidChangeSelection(async ([{ label }]) => {
+      quickPick.hide();
+
+      await longRunning(`Generate SAS for ${clusterName} cluster.`,
```

Comment:
I don't get why you set up the quick pick, then do this long-running thing, _then_ show the quick pick UI.

Reply:
💡 I am easy with any simpler mechanism to keep user updated, main intent is to inform user that -- since you have selected your account in case of more than 1 account, code will go and generate the SAS. along with few other long running tasks.